compiler/luci-interpreter/include/luci_interpreter/Interpreter.h

```diff
@@ -48,10 +48,11 @@ class Interpreter
 
 private:
   void createTensors(const loco::Graph *graph);
-  void createExecutionSequence(const loco::Graph *main_graph);
+  void createKernels(const loco::Graph *graph);
```

Comment:
`createKernels` is a private function called inside the constructor. This does not allow developers to install custom kernels at the interpreter (Am I right?).

Then, how about making a public function that sets `_kernel_map` to the KernelMap given by a developer ? If so, it will be possible for developers to make their own KernelMap and build the interpreter with it.

Reply:
This is not quite easy thing to do. I considered it when was preparing draft, but decided not to do it. Is this functionality really needed?

Reply:
> Is this functionality really needed?

What we need is to call the function that records the tensor data inside the `interpret` function (after `kernel->execute()`). I thought that the above functionality, which is similar to TFLite, enables it (1). 

(2) Another way would be to initialize the interpreter with a modified graph (where a new node that profiles tensor data is inserted after each CircleNode). In this case, we will need to implement a special loco node for profiling tensor data and rewrite loco graph.

(3) I've talked with @seanshpark and @parjong , and they recommended to use an observer pattern. That is, we call an observer function (which observing tensor data) in the interpret loop after kernel is executed. This will change the interpret loop, but it seems to be the simplest way to implement profiling functionality.

What do you think of the above methods (1, 2, 3)?

Reply:
> What we need is to call the function that records the tensor data inside the interpret function (after kernel->execute()). I thought that the above functionality (similar thing is supported in TFLite) enables it (1).

Oh, I see. I don't know why TFLite has it this way, may be due to historical reasons. I think we don't need to follow them.

> (2) Another way would be to initialize the interpreter with a modified graph (where a new node that profiles tensor data is inserted after each CircleNode). In this case, we will need to implement a special loco node for profiling tensor data and rewrite loco graph.

The same thing is done in armnn. It is an interesting approach, but in the essense it is the same as (3), except that requires more work.

> (3) I've talked with @seanshpark and @parjong , and they mentioned to use an observer pattern. That is, we call an observer function (which observing tensor data) in the interpret loop after each kernel is executed. This will change the interpret loop, but it seems to be the simplest way to implement profiling functionality.

This is what I was going to do :) Call the observer function (== hook == callback in some terminology) in the interpreter loop. The function will have the node and the output tensor(s) as parameters, and it will be responsible for doing whatever it wants with that data.

However, it is out of scope of this PR, which is just to make the "node" available in the interpreter loop. I suggest to merge this one and continue discussion in more details in #1098.